src/app/admin/group/group-modal.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { GroupAddEditDialogComponent } from './group-add-edit-dialog/group-add-edit-dialog.component';
+import { GroupViewDialogComponent } from './group-view-dialog/group-view-dialog.component';
+import { MatDialog } from '@angular/material';
+import { DialogData } from './../group/group-modal.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GroupModalService {
+
+  constructor(public dialog: MatDialog) { }
+
+  // create modal window for add new group or edit group
+  addGroupDialog(data: DialogData, callBackFunction: Function): void {
+    const dialogRef = this.dialog.open(GroupAddEditDialogComponent, {
```

Comment:
Here is three methods with a similar code. Can it be refactored to avoid such duplication?

Reply:
Three methods were replaced by one